In [3]:
lienbdd = "https://raw.githubusercontent.com/agailloty/algopy-2023/main/projets/ML/TravelInsurancePrediction.csv"

In [5]:
import pandas as pd
travel_data = pd.read_csv(lienbdd)

In [6]:
travel_data.head(6)

Unnamed: 0  Age               Employment Type GraduateOrNot  AnnualIncome  \
0           0   31             Government Sector           Yes        400000   
1           1   31  Private Sector/Self Employed           Yes       1250000   
2           2   34  Private Sector/Self Employed           Yes        500000   
3           3   28  Private Sector/Self Employed           Yes        700000   
4           4   28  Private Sector/Self Employed           Yes        700000   
5           5   25  Private Sector/Self Employed            No       1150000   

   FamilyMembers  ChronicDiseases FrequentFlyer EverTravelledAbroad  \
0              6                1            No                  No   
1              7                0            No                  No   
2              4                1            No                  No   
3              3                1            No                  No   
4              8                1           Yes                  No   
5              4                0            No                  No   

   TravelInsurance  
0                0  
1                0  
2                1  
3                0  
4                0  
5                0

In [7]:
travel_data.tail(5)

Unnamed: 0  Age               Employment Type GraduateOrNot  \
1982        1982   33  Private Sector/Self Employed           Yes   
1983        1983   28  Private Sector/Self Employed           Yes   
1984        1984   28  Private Sector/Self Employed           Yes   
1985        1985   34  Private Sector/Self Employed           Yes   
1986        1986   34  Private Sector/Self Employed           Yes   

      AnnualIncome  FamilyMembers  ChronicDiseases FrequentFlyer  \
1982       1500000              4                0           Yes   
1983       1750000              5                1            No   
1984       1150000              6                1            No   
1985       1000000              6                0           Yes   
1986        500000              4                0            No   

     EverTravelledAbroad  TravelInsurance  
1982                 Yes                1  
1983                 Yes                0  
1984                  No                0  
1985                 Yes                1  
1986                  No                0

In [8]:
# Eplorer la base de données
travel_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1987 entries, 0 to 1986
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Unnamed: 0           1987 non-null   int64 
 1   Age                  1987 non-null   int64 
 2   Employment Type      1987 non-null   object
 3   GraduateOrNot        1987 non-null   object
 4   AnnualIncome         1987 non-null   int64 
 5   FamilyMembers        1987 non-null   int64 
 6   ChronicDiseases      1987 non-null   int64 
 7   FrequentFlyer        1987 non-null   object
 8   EverTravelledAbroad  1987 non-null   object
 9   TravelInsurance      1987 non-null   int64 
dtypes: int64(6), object(4)
memory usage: 155.4+ KB


In [9]:
travel_data.describe()

Unnamed: 0          Age  AnnualIncome  FamilyMembers  ChronicDiseases  \
count  1987.000000  1987.000000  1.987000e+03    1987.000000      1987.000000   
mean    993.000000    29.650226  9.327630e+05       4.752894         0.277806   
std     573.741812     2.913308  3.768557e+05       1.609650         0.448030   
min       0.000000    25.000000  3.000000e+05       2.000000         0.000000   
25%     496.500000    28.000000  6.000000e+05       4.000000         0.000000   
50%     993.000000    29.000000  9.000000e+05       5.000000         0.000000   
75%    1489.500000    32.000000  1.250000e+06       6.000000         1.000000   
max    1986.000000    35.000000  1.800000e+06       9.000000         1.000000   

       TravelInsurance  
count      1987.000000  
mean          0.357323  
std           0.479332  
min           0.000000  
25%           0.000000  
50%           0.000000  
75%           1.000000  
max           1.000000

In [10]:
travel_data.describe(include = ["object"])

Employment Type GraduateOrNot FrequentFlyer  \
count                           1987          1987          1987   
unique                             2             2             2   
top     Private Sector/Self Employed           Yes            No   
freq                            1417          1692          1570   

       EverTravelledAbroad  
count                 1987  
unique                   2  
top                     No  
freq                  1607

Dans la colonne Employment Type la modalité Private Sector/Self Employed revient 1417 fois sur 1987.

In [12]:
# Des statistiques groupées
travel_data.groupby("GraduateOrNot").describe()["ChronicDiseases"]

count      mean       std  min  25%  50%  75%  max
GraduateOrNot                                                     
No              295.0  0.257627  0.438071  0.0  0.0  0.0  1.0  1.0
Yes            1692.0  0.281324  0.449778  0.0  0.0  0.0  1.0  1.0

Statistiques groupées
Il y a 1692 personnes qui ont un diplôme de Bac +5

In [13]:
# Vérifier s'il n'y a pas de données manquantes dans le jeu de données
travel_data.isna().sum()

Unnamed: 0             0
Age                    0
Employment Type        0
GraduateOrNot          0
AnnualIncome           0
FamilyMembers          0
ChronicDiseases        0
FrequentFlyer          0
EverTravelledAbroad    0
TravelInsurance        0
dtype: int64

Bonne nouvelle, pas de données manquantes. On gagne du temps !

In [24]:
travel_data.groupby("Employment Type").agg(["count", "mean", "median"])["FamilyMembers"]

count      mean  median
Employment Type                                      
Government Sector               570  4.761404     5.0
Private Sector/Self Employed   1417  4.749471     5.0

# Entrainement des modèles
On aimerait estimer la probabilité que la personne ait souscrit à une police d'assurance. 

In [25]:
# Lister toutes les colonnes du jeu de données
travel_data.columns

Index(['Unnamed: 0', 'Age', 'Employment Type', 'GraduateOrNot', 'AnnualIncome',
       'FamilyMembers', 'ChronicDiseases', 'FrequentFlyer',
       'EverTravelledAbroad', 'TravelInsurance'],
      dtype='object')

In [28]:
df = pd.get_dummies(travel_data, columns = ['Employment Type', 'GraduateOrNot', 'EverTravelledAbroad', 'FrequentFlyer'], drop_first=True)

In [31]:
df.pop("Unnamed: 0")

0          0
1          1
2          2
3          3
4          4
        ... 
1982    1982
1983    1983
1984    1984
1985    1985
1986    1986
Name: Unnamed: 0, Length: 1987, dtype: int64

In [30]:
# Il faut scinder le jeu de données : en jeu de donnée d'entrainement et jeu de données test

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1987 entries, 0 to 1986
Data columns (total 9 columns):
 #   Column                                        Non-Null Count  Dtype
---  ------                                        --------------  -----
 0   Age                                           1987 non-null   int64
 1   AnnualIncome                                  1987 non-null   int64
 2   FamilyMembers                                 1987 non-null   int64
 3   ChronicDiseases                               1987 non-null   int64
 4   TravelInsurance                               1987 non-null   int64
 5   Employment Type_Private Sector/Self Employed  1987 non-null   uint8
 6   GraduateOrNot_Yes                             1987 non-null   uint8
 7   EverTravelledAbroad_Yes                       1987 non-null   uint8
 8   FrequentFlyer_Yes                             1987 non-null   uint8
dtypes: int64(5), uint8(4)
memory usage: 85.5 KB


In [34]:
from sklearn.model_selection import train_test_split

In [36]:
y = df.pop("TravelInsurance") # Supprimer la colonne
X = df

In [37]:
X.head()

Age  AnnualIncome  FamilyMembers  ChronicDiseases  \
0   31        400000              6                1   
1   31       1250000              7                0   
2   34        500000              4                1   
3   28        700000              3                1   
4   28        700000              8                1   

   Employment Type_Private Sector/Self Employed  GraduateOrNot_Yes  \
0                                             0                  1   
1                                             1                  1   
2                                             1                  1   
3                                             1                  1   
4                                             1                  1   

   EverTravelledAbroad_Yes  FrequentFlyer_Yes  
0                        0                  0  
1                        0                  0  
2                        0                  0  
3                        0                  0  
4                        0                  1

In [38]:
y

0       0
1       0
2       1
3       0
4       0
       ..
1982    1
1983    0
1984    0
1985    1
1986    0
Name: TravelInsurance, Length: 1987, dtype: int64

In [39]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

In [40]:
print(f"Le jeu de données d'entraînement contient {X_train.shape[0]} lignes et {X_train.shape[1]} colonnes ")

Le jeu de données d'entraînement contient 1331 lignes et 8 colonnes 


In [ ]:
#